src/src/Helpers/ESPEasy_TouchHandler.cpp

```diff
@@ -1706,14 +1679,26 @@ bool ESPEasy_TouchHandler::plugin_fifty_per_second(struct EventStruct *event,
 
       // Do NOT send a Z event for each touch?
       if (!bitRead(Touch_Settings.flags, TOUCH_FLAGS_SEND_Z) && validDeviceIndex(DeviceIndex)) {
+        // FIXME TD-er: Should not change anything in the Device vector.
+        # ifdef ESP8266
         Device[DeviceIndex].VType      = Sensor_VType::SENSOR_TYPE_DUAL;
         Device[DeviceIndex].ValueCount = 2;
+        # else // ifdef ESP8266
+        Device.getDeviceStructForEdit(DeviceIndex).VType      = Sensor_VType::SENSOR_TYPE_DUAL;
```

Comment:
That function was there as a reminder not to use it....
Device vector should be considered const.

Reply:
Yup, understood, but there currently isn't a feature available to have variation in the number of values to send based on task settings, AFAICS, so this is the least worse solution.
NB: Copied that from P099 😋

Reply:
Ah check, so that's up to me to do some work ;)

Reply:
Or it could be implemented using the `PLUGIN_GET_DEVICEVTYPE` plugin event. Currently it's changed 'on the fly', just before sending the data to the controller, but that shouldn't be needed once set via the event 🤔.

Reply:
Problem with non-const Devices array is that such changes aren't task specific.

Reply:
I'm currently switching to `PLUGIN_GET_DEVICEVALUECOUNT`, as that is actually the _correct_ place to set the number of output values for this plugin. No need to change the `Device_VType` setting, so that was in fact a bug 😞